.github/workflows/integration_tests.yml

```diff
@@ -63,42 +67,38 @@ jobs:
       - name: Set env variables for subsequent steps (all)
         run: |
           echo "::set-env name=VCPKG_RESPONSE_FILE::external/vcpkg_${{ matrix.vcpkg_triplet }}_response_file.txt"
-          echo "::set-env name=MATRIX_UNIQUE_NAME::${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.architecture }}-${{ matrix.python_version }}"
       - name: Cache vcpkg C++ dependencies
+        if: matrix.target_platform == 'Desktop'
         id: cache_vcpkg
         uses: actions/cache@v2
         with:
           path: external/vcpkg/installed
           key: dev-vcpkg-${{ matrix.vcpkg_triplet }}-${{ hashFiles(format('{0}', env.VCPKG_RESPONSE_FILE)) }}-${{ hashFiles('.git/modules/external/vcpkg/HEAD') }}
 
-      - name: Cache ccache files
-        if: startsWith(matrix.os, 'ubuntu') || startsWith(matrix.os, 'macos')
-        id: cache_ccache
-        uses: actions/cache@v2
-        with:
-          path: ccache_dir
-          key: dev-test-ccache-${{ env.MATRIX_UNIQUE_NAME }}
-
       - name: Setup python
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python_version }}
           architecture: ${{ matrix.architecture }}
 
-      - name: Install SDK prerequisites
+      - name: Install SDK Desktop prerequisites
+        if: matrix.target_platform == 'Desktop'
         run: |
           python scripts/gha/install_prereqs_desktop.py
           
+      - name: Install SDK Android prerequisites
+        if: matrix.target_platform == 'Android'
+        run: |
+          build_scripts/android/install_prereqs.sh
+          
       - name: Prepare for integration tests
         run: |
           pip install -r testing/integration_testing/requirements.txt
-          python testing/integration_testing/restore_secrets.py --passphrase ${{ secrets.TEST_SECRET }}
+          python testing/integration_testing/restore_secrets.py --passphrase "${{ secrets.TEST_SECRET }}"
           
       - name: Build and run integration tests
+        # The set up script for Android will download the NDK here.
+        env:
+          NDK_ROOT: '/tmp/android-ndk-r16b'
```

Comment:
Is this workflow supported on Windows? Could this cause an issue?

Reply:
It's not supported on Windows, no. The SDK build workflow for Android is only for Linux and MacOS at the moment, so this matches that.

The build setup script downloads the NDK and puts it in this directory, so that's why I'm setting this env var accordingly.  When the SDK build workflow is updated to also work on Windows I'll update this to use whatever solution is used for that.

Reply:
The android prereqs script I wrote already fetches NDK-r16b in the script itself. (it's still bash using curl, but will be switched to Python asap). We should just do that, not use additional workflows - as little as possible should be in the yml files.

Reply:
Oh I misunderstood this, this is just using the NDK that the script has already downloaded. Never mind. But yes it will be updated to Python and support Windows soon.